drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2077 @@
+
+	LOG_DBG("cache read type %d len %d", reg_type, size);
+
+	ret = gpio_pca_series_reg_cache_get_offset(dev, reg_type, &offset);
+	if (ret != 0) {
+		LOG_ERR("can not get noncacheable reg");
+		ret = -EACCES;
+		goto out;
+	}
+	memcpy(buf, ((uint8_t *)&data->cache) + offset, size);
+
+out:
+	LOG_DBG("cache read return %d", ret);
+	return ret;
+}
+
+/**
+ * @brief update register cache from device or existing value.
+ *
+ * @param dev
+ * @param reg_type
+ * @param buf pointer to new data to update from. if NULL, read reg value from device
+ * @return int 0        if success
+ *             -EACCES  if register is uncacheable
+ *             -EFAULT  if register is unsupported when trying to read
+ */
+static inline int gpio_pca_series_reg_update_cache(const struct device *dev,
+						   enum gpio_pca_series_reg_type reg_type,
+						   uint8_t *buf)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	struct gpio_pca_series_data *data = dev->data;
+	int ret = 0;
+	uint32_t offset;
+	uint32_t size = gpio_pca_series_reg_get_size(dev, reg_type);
+	struct i2c_msg msg[2];
+
+	LOG_DBG("cache update type %d len %d", reg_type, size);
+
+	ret = gpio_pca_series_reg_cache_get_offset(dev, reg_type, &offset);
+	if (ret != 0) {
+		LOG_ERR("can not update noncacheable reg");
+		return -EACCES;
+	}
+	if (buf != NULL) {
+		/** update cache from buf */
+		memcpy(buf, ((uint8_t *)&data->cache) + offset, size);
+		goto out;
+	}
+
+	uint8_t addr = gpio_pca_series_reg_get_addr(dev, reg_type);
+
+	LOG_INF("reg read type %d addr 0x%x len %d", reg_type, addr, size);
+
+	if (addr == PCA_REG_INVALID) {
+		LOG_ERR("trying to read unsupported reg");
+		ret = -EFAULT;
+		goto out;
+	}
+
+	msg[0].buf = &addr;
+	msg[0].len = 1;
+	msg[0].flags = I2C_MSG_WRITE;
+	msg[1].buf = buf;
+	msg[1].len = size;
+	msg[1].flags = I2C_MSG_RESTART | I2C_MSG_READ | I2C_MSG_STOP;
+
+	ret =  i2c_transfer(dev, msg, 2, addr);
+	if (ret) {
+		LOG_ERR("i2c read error [%d]", ret);
+		goto out;
+	}
+
+out:
+	LOG_DBG("cache read return %d", ret);
+	return ret;
+}
+
+#else /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+#define gpio_pca_series_reg_read_cache gpio_pca_series_reg_read
+
+struct gpio_pca_series_reg_cache_mini {
+	uint32_t output;   /** cache output value for faster output */
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	uint32_t input;    /** only used when interrupt mask & edge config is not present */
+	uint32_t int_rise; /** only used if interrupt edge is software-compared */
+	uint32_t int_fall; /** only used if interrupt edge is software-compared */
+#endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */
+};
+
+/**
+ * @}
+ * // gpio_pca_cache_api
+ */
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+/*!
+ * @addtogroup gpio_pca_custom_api
+ * @{
+ */
+
+int gpio_pca_series_reset(const struct device *dev)
+{
+	/* TODO assert reset pin if available */
+	/* TODO or write reset value */
+	return -ENOTSUP;
+}
+
+void gpio_pca_series_dump(const struct device *dev)
```

Comment:
Does not appear this function is used- I understand needing it for debugging, but perhaps we could add somewhere where the function is called if `CONFIG_GPIO_LOG_LEVEL_DBG` is set? Otherwise, this is effectively dead code

Reply:
This still needs to be resolved

Reply:
Added KConfig option to dump all registers after init.